lib/puppet/module.rb

```diff
@@ -180,7 +180,7 @@ def match_manifests(rest)
 
     # (#4220) Always ensure init.pp in case class is defined there.
     init_manifest = manifest("init.pp")
-    if !init_manifest.nil? && !searched_manifests.include?(init_manifest)
+    if !init_manifest.nil? && !searched_manifests.include?(Dir.glob(init_manifest).first)
```

Comment:
One thought here is that we could pay the tax only on Windows, by using a method on `Puppet::FileSystem` here to expand the path out.